libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpVP9Reader.java

```diff
@@ -0,0 +1,230 @@
+ * limitations under the License.
+ */
+package androidx.media3.exoplayer.rtsp.reader;
+
+import static androidx.media3.common.util.Assertions.checkArgument;
+import static androidx.media3.common.util.Assertions.checkStateNotNull;
+import static androidx.media3.common.util.Util.castNonNull;
+
+import androidx.media3.common.C;
+import androidx.media3.common.ParserException;
+import androidx.media3.common.util.Log;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.common.util.Util;
+import androidx.media3.exoplayer.rtsp.RtpPacket;
+import androidx.media3.exoplayer.rtsp.RtpPayloadFormat;
+import androidx.media3.extractor.ExtractorOutput;
+import androidx.media3.extractor.TrackOutput;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+
+/**
+ * Parses an VP9 byte stream carried on RTP packets, and extracts VP9 Access Units. Refer to
+ * @link https://datatracker.ietf.org/doc/html/draft-ietf-payload-vp9 for more details.
+ */
+/* package */ final class RtpVP9Reader implements RtpPayloadReader {
+
+  private static final String TAG = "RtpVP9Reader";
+
+  private static final long MEDIA_CLOCK_FREQUENCY = 90_000;
+
+  private final RtpPayloadFormat payloadFormat;
+
+  private @MonotonicNonNull TrackOutput trackOutput;
+  @C.BufferFlags private int bufferFlags;
+
+  private long firstReceivedTimestamp;
+  private long startTimeOffsetUs;
+  private static int previousSequenceNumber;
+  /** The combined size of a sample that is fragmented into multiple RTP packets. */
+  private int fragmentedSampleSizeBytes;
+  private static int width;
+  private static int height;
+  private static boolean gotFirstPacketOfVP9Frame;
+  private boolean isKeyFrame;
+  private boolean isOutputFormatSet;
+
+  /** Creates an instance. */
+  public RtpVP9Reader(RtpPayloadFormat payloadFormat) {
+    this.payloadFormat = payloadFormat;
+    firstReceivedTimestamp = C.TIME_UNSET;
+    startTimeOffsetUs = 0;
+    previousSequenceNumber = C.INDEX_UNSET;
+    fragmentedSampleSizeBytes = 0;
+    width = C.INDEX_UNSET;
+    height = C.INDEX_UNSET;
+    gotFirstPacketOfVP9Frame = false;
+    isKeyFrame = false;
+    isOutputFormatSet = false;
+  }
+
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_VIDEO);
+    castNonNull(trackOutput).format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {}
+
+  @Override
+  public void consume(ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker)
+      throws ParserException {
+    checkStateNotNull(trackOutput);
+
+    if (parseVP9Descriptor(data, sequenceNumber)) {
+      if (fragmentedSampleSizeBytes == 0 && gotFirstPacketOfVP9Frame) {
+        isKeyFrame = (data.peekUnsignedByte() & 0x04) == 0;
+      }
+
+      if (!isOutputFormatSet && width > 0 && height > 0) {
```

Comment:
You initialized width/height to `UNSET`, you can check it using `UNSET` here like `width != UNSET`, rather than `width > 0`.